test/test_format.py

```diff
@@ -8,523 +8,789 @@
+    },
+    "-4.32`4": {
+        "msg": "A PrecisionReal number",
+        "text": {
+            "System`StandardForm": "-4.320",
+            "System`TraditionalForm": "-4.320",
+            "System`InputForm": "-4.320",
+            "System`OutputForm": "-4.320",
+        },
+        "mathml": {
+            "System`StandardForm": (
+                "<mn>-4.320</mn>",
+                "must be <mn>, not <mtext>",
+            ),
+            "System`TraditionalForm": (
+                "<mn>-4.320</mn>",
+                "must be <mn>, not <mtext>",
+            ),
+            "System`InputForm": "<mtext>-4.320</mtext>",
+            "System`OutputForm": "<mtext>-4.320</mtext>",
+        },
+        "tex": {
+            "System`StandardForm": "-4.320",
+            "System`TraditionalForm": "-4.320",
+            "System`InputForm": "-4.320",
+            "System`OutputForm": "-4.320",
+        },
+    },
+    "-4.326563712`2": {
+        "msg": "A Real number",
+        "text": {
+            "System`StandardForm": "-4.3",
+            "System`TraditionalForm": "-4.3",
+            "System`InputForm": "-4.3",
+            "System`OutputForm": "-4.3",
+        },
+        "mathml": {
+            "System`StandardForm": (
+                "<mn>-4.3</mn>",
+                "must be <mn>, not <mtext>",
+            ),
+            "System`TraditionalForm": (
+                "<mn>-4.3</mn>",
+                "must be <mn>, not <mtext>",
+            ),
+            "System`InputForm": "<mtext>-4.3</mtext>",
+            "System`OutputForm": "<mtext>-4.3</mtext>",
+        },
+        "tex": {
+            "System`StandardForm": "-4.3",
+            "System`TraditionalForm": "-4.3",
+            "System`InputForm": "-4.3",
+            "System`OutputForm": "-4.3",
+        },
+    },
+    '"Hola!"': {
+        "msg": "A String",
+        "text": {
+            "System`StandardForm": "Hola!",
+            "System`TraditionalForm": "Hola!",
+            "System`InputForm": "Hola!",
+            "System`OutputForm": "Hola!",
+        },
+        "mathml": {
+            "System`StandardForm": "<mtext>Hola!</mtext>",
+            "System`TraditionalForm": "<mtext>Hola!</mtext>",
+            "System`InputForm": "<mtext>Hola!</mtext>",
+            "System`OutputForm": "<mtext>Hola!</mtext>",
+        },
+        "tex": {
+            "System`StandardForm": "\\text{Hola!}",
+            "System`TraditionalForm": "\\text{Hola!}",
+            "System`InputForm": "\\text{Hola!}",
+            "System`OutputForm": "\\text{Hola!}",
+        },
+    },
+    # String with special characters
+    '"\\[Pi] is a trascendent number"': {
+        "msg": "A String",
+        "text": {
+            "System`StandardForm": "π is a trascendent number",
+            "System`TraditionalForm": "π is a trascendent number",
+            "System`InputForm": "π is a trascendent number",
+            "System`OutputForm": "π is a trascendent number",
+        },
```

Comment:
All of the test above don't seem that fragile. I is just when we start getting to stuff like 

```
"<mtext>π&nbsp;is&nbsp;a&nbsp;trascendent&nbsp;number</mtext>",
``` 

or worse: 

````
"System`StandardForm": '<mrow><msubsup><mo>∫</mo> <mi>a</mi> <mrow><mi>g</mi> <mo>[</mo> <mi>b</mi> <mo>]</mo></mrow></msubsup> <mo form="prefix" lspace="0" rspace="0.2em">\u2062</mo> <mrow><mi>F</mi> <mo>[</mo> <mi>x</mi> <mo>]</mo></mrow> <mo form="prefix" lspace="0" rspace="0.2em">\u2062</mo> <mrow><mtext>\uf74c</mtext> <mi>x</mi></mrow></mrow>',
            "System`TraditionalForm": '<mrow><msubsup><mo>∫</mo> <mi>a</mi> <mrow><mi>g</mi> <mo>(</mo> <mi>b</mi> <mo>)</mo></mrow></msubsup> <mo form="prefix" lspace="0" rspace="0.2em">\u2062</mo> <mrow><mi>F</mi> <mo>(</mo> <mi>x</mi> <mo>)</mo></mrow> <mo form="prefix" lspace="0" rspace="0.2em">\u2062</mo> <mrow><mtext>\uf74c</mtext> <mi>x</mi></mrow></mrow>',
            "System`InputForm": "<mrow><mi>Integrate</mi> <mo>[</mo> <mrow><mrow><mi>F</mi> <mo>[</mo> <mi>x</mi> <mo>]</mo></mrow> <mtext>,&nbsp;</mtext> <mrow><mo>{</mo> <mrow><mi>x</mi> <mtext>,&nbsp;</mtext> <mi>a</mi> <mtext>,&nbsp;</mtext> <mrow><mi>g</mi> <mo>[</mo> <mi>b</mi> <mo>]</mo></mrow></mrow> <mo>}</mo></mrow></mrow> <mo>]</mo></mrow>",
            "System`OutputForm": "<mrow><mi>Integrate</mi> <mo>[</mo> <mrow><mrow><mi>F</mi> <mo>[</mo> <mi>x</mi> <mo>]</mo></mrow> <mtext>,&nbsp;</mtext> <mrow><mo>{</mo> <mrow><mi>x</mi> <mtext>,&nbsp;</mtext> <mi>a</mi> <mtext>,&nbsp;</mtext> <mrow><mi>g</mi> <mo>[</mo> <mi>b</mi> <mo>]</mo></mrow></mrow> <mo>}</mo></mrow></mrow> <mo>]</mo></mrow>
```
or even more bad:

```
System`StandardForm": '<mglyph width="294px" height="350px" src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjEuMHB4IiBoZW... 
``` 

where it is annoying. (Personally, I dislike all the extra work this is going to cause me/us when thing change, but I guess I'll just live with it.)

But  using your new-found skills of trying to find the _smallest_ example that exhibits the feature you want to test would be most appreciated here.